judge/runpipe.cc

```diff
@@ -846,6 +913,18 @@ struct state_t {
           }
           continue;
         }
+        if (fd == child_timelimit_pipe) {
+            static char buffer[1];
+            if (read(child_timelimit_pipe, buffer, 1) != 1) {
```

Comment:
Would be good to also check that `buffer` contains 42. Although I see it's also not done for the code you copy&pasted from.

Reply:
Done